server/src/main.rs

```diff
@@ -157,57 +133,41 @@ async fn init_grpc_clients() {
     let storage_full_grpc_addr =
         format!("http://{storage_grpc_host}:{storage_grpc_port}").to_string();
 
-    match FLIGHT_PLAN_CLIENT
-        .get_or_try_init(|| async {
-            println!(
-                "Setting up connection to svc-storage flight plan on {}",
-                storage_full_grpc_addr
-            );
-            FlightPlanRpcClient::connect(storage_full_grpc_addr.clone()).await
-        })
-        .await
+    info!(
+        "Setting up connection to svc-storage clients on {}",
+        storage_full_grpc_addr.clone()
+    );
+    let flight_plan_client_res = FlightPlanRpcClient::connect(storage_full_grpc_addr.clone()).await;
+    let vehicle_client_res = VehicleRpcClient::connect(storage_full_grpc_addr.clone()).await;
+    let vertiport_client_res = VertiportRpcClient::connect(storage_full_grpc_addr.clone()).await;
+    let vertipad_client_res = VertipadRpcClient::connect(storage_full_grpc_addr.clone()).await;
+    if flight_plan_client_res.is_err()
+        || vehicle_client_res.is_err()
+        || vertiport_client_res.is_err()
+        || vertipad_client_res.is_err()
     {
-        Ok(_) => (),
-        Err(e) => println!(
-            "Unable to connect to svc-storage flight plan at {}; {}",
+        error!(
+            "Failed to connect to storage service at {}. Client errors: {} {} {} {}",
             storage_full_grpc_addr.clone(),
-            e
-        ),
-    };
-
-    match VERTIPORT_CLIENT
-        .get_or_try_init(|| async {
-            println!(
-                "Setting up connection to svc-storage vertiport on {}",
-                storage_full_grpc_addr
-            );
-            VertiportRpcClient::connect(storage_full_grpc_addr.clone()).await
-        })
-        .await
-    {
-        Ok(_) => (),
-        Err(e) => println!(
-            "Unable to connect to svc-storage vertiport at {}; {}",
-            storage_full_grpc_addr, e
-        ),
-    };
-
-    match VEHICLE_CLIENT
-        .get_or_try_init(|| async {
-            println!(
-                "Setting up connection to svc-storage vehicle on {}",
-                storage_full_grpc_addr
-            );
-            VehicleRpcClient::connect(storage_full_grpc_addr.clone()).await
-        })
-        .await
-    {
-        Ok(_) => (),
-        Err(e) => println!(
-            "Unable to connect to svc-storage vehicle at {}; {}",
-            storage_full_grpc_addr, e
-        ),
-    };
+            flight_plan_client_res.err().unwrap(),
+            vehicle_client_res.err().unwrap(),
+            vertiport_client_res.err().unwrap(),
+            vertipad_client_res.err().unwrap()
+        );
+        panic!();
```

Comment:
If some of the storgage grpc clients fail, then this is non-recoverable error and we should just stop the service - panic does that. Results are used for recoverable errors - like if during one request something fails, we would return/log the error but the service continues to work and receive requests.

Reply:
What would be the protocol for restarting the server then? Are we expecting manual restarts?

Reply:
We're running everything in Docker. It will be restarted automatically because the health checks will fail. 
I agree that we should panic here, this way we'll be able to see in our monitoring that something is wrong in our network and respond to it accordingly.